libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -608,11 +612,13 @@ method publish*(g: GossipSub,
   return peers.len
 
 proc maintainDirectPeer(g: GossipSub, id: PeerId, addrs: seq[MultiAddress]) {.async.} =
-  let peer = g.peers.getOrDefault(id)
-  if isNil(peer):
+  if id notin g.peers:
     trace "Attempting to dial a direct peer", peer = id
+    if g.switch.isConnected(id):
+      info "We are connected to a direct peer, but GossipSub isn't setup with him!", id
```

Comment:
```suggestion
      info "We are connected to a direct peer, but it isn't a GossipSub peer!", id
```

Reply:
Btw should we do something more in this case? If it's unexpected, maybe it should be a warn.

Reply:
It's basically a user error, but not critical, so info seems like a good fit

Reply:
I'd use debug or warn.